tiledb/subarray.py

```diff
@@ -0,0 +1,80 @@
+from numbers import Real
+from typing import (
+    TYPE_CHECKING,
+    Sequence,
+    Tuple,
+)
+
+import numpy as np
+
+import tiledb.cc as lt
+
+from ._array import ArrayImpl
+from .ctx import CtxMixin
+from .libtiledb import Array
+
+if TYPE_CHECKING:
+    from libtiledb import Ctx
+
+
+Scalar = Real
+Range = Tuple[Scalar, Scalar]
+
+
+class Subarray(CtxMixin, lt.Subarray):
+    """
+    Represents a TileDB subarray.
+    """
+
+    def __init__(
+        self,
+        array: Array,
+        ctx: "Ctx" = None,
+    ):
+        """Class representing a subarray of a TileDB Array.
+
+        :param array: tiledb.Array the subarray is defined on
+        :param ctx: A TileDB Context
+        """
+        self._array = array
+        self._cpp_array = ArrayImpl(array)
+        super().__init__(ctx, self._cpp_array)
```

Comment:
The `_cpp_array` attribute is referenced only here, any reason to store it instead of `super().__init__(ctx, ArrayImpl(array))`?